File: Correct/2학년/D. 조별과제 팀뽑기/main.py
N, X = map(int, input().split())
scores = list(map(int, input().split()))

scores.sort()

left = 0
right = N - 1
count = 0

while left < right:
    s = scores[left] + scores[right]
    if s == X:
        count += 1
        left += 1
        right -= 1
    elif s < X:
        left += 1
    else:
        right -= 1

print(count)

File: Correct/2학년/D. 조별과제 팀뽑기/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N;
    long long X;
    cin >> N >> X;

    vector<int> scores(N);
    for (int i = 0; i < N; ++i)
        cin >> scores[i];

    sort(scores.begin(), scores.end());

    int left = 0, right = N - 1, count = 0;

    while (left < right) {
        long long sum = (long long)scores[left] + scores[right];
        if (sum == X) {
            ++count;
            ++left;
            --right;
        } else if (sum < X) {
            ++left;
        } else {
            --right;
        }
    }

    cout << count << '\n';
    return 0;
}